### PR TITLE
chore(deps): bump various dependencies

### PR DESCRIPTION
**Pull request by RTann**
## Description

Update various packages to the version the current `master` branch uses.

Highlights:

* gopkg.in/square/go-jose.v2 v2.6.0 => github.com/go-jose/go-jose/v3 v3.0.3
* github.com/openshift/api v0.0.0-20230815081605-d7f5a4c98505 => github.com/openshift/api v0.0.0-20231117201702-2ea16bbab164
* github.com/sigstore/cosign/v2 v2.0.2 => github.com/sigstore/cosign/v2 v2.2.2
* helm.sh/helm/v3 v3.13.2 => helm.sh/helm/v3 v3.14.2
* k8s.io/api v0.28.3 => k8s.io/api v0.29.2
* github.com/containerd/containerd v1.7.6 // indirect => github.com/containerd/containerd v1.7.11 // indirect
* google.golang.org/protobuf v1.31.0 // indirect => google.golang.org/protobuf v1.33.0 // indirect

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

### Here I tell how I validated my change

If it builds, it builds

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by rukletsov** [comment]
I'm puzzled how come newer version of k8s.api works withoug bumping np-guard dependencies, see https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/flowcontrol/v1alpha1:
```
→ go mod tidy
go: finding module for package k8s.io/api/flowcontrol/v1alpha1
github.com/stackrox/rox/roxctl/netpol/generate imports
	github.com/np-guard/cluster-topology-analyzer/pkg/controller imports
	github.com/openshift/api imports
	k8s.io/api/flowcontrol/v1alpha1: module k8s.io/api@latest found (v0.29.2), but does not contain package k8s.io/api/flowcontrol/v1alpha1
```

File: go.mod [hunk 1]
```diff
@@ -96,12 +97,12 @@ require (
 	github.com/russellhaering/goxmldsig v1.4.0
 	github.com/segmentio/analytics-go/v3 v3.2.1
 	github.com/sergi/go-diff v1.3.1
-	github.com/sigstore/cosign/v2 v2.0.2
+	github.com/sigstore/cosign/v2 v2.2.2
```
Comment (janisz) [inline go.mod:100]: Why it was changed? We have a replace for cosign. Do we need to rebase our fork? https://github.com/stackrox/stackrox/blob/5f3cfdea3d40b8a09cca0a9bc968711bd2041e00/go.mod#L460
cc @dhaus67

File: go.mod [hunk 2]
```diff
@@ -123,28 +124,27 @@ require (
...
-	k8s.io/client-go v0.28.3
-	k8s.io/kubectl v0.28.2
+	helm.sh/helm/v3 v3.14.2
+	k8s.io/api v0.29.2
+	k8s.io/apiextensions-apiserver v0.29.0
+	k8s.io/apimachinery v0.29.2
+	k8s.io/apiserver v0.29.0
+	k8s.io/cli-runtime v0.29.1
+	k8s.io/client-go v0.29.1
+	k8s.io/kubectl v0.29.1
```
Comment (janisz) [inline go.mod:145]: Why not keep everything on 0.29.2?
Also I think this deserves a separated PR

File: go.mod (hunk 1 above)
Comment (vladbologa) [reply go.mod:100]: Don't know why, but also `master` and `release-4.4` use `github.com/sigstore/cosign/v2 v2.2.2`

**Comment by RTann** [comment]
I'm going to close this in favor of multiple, simpler PRs

**Pull request closed without merging**